core/src/com/unciv/models/ruleset/Unique.kt

```diff
@@ -126,10 +312,200 @@ object UniqueTriggerActivation {
+                val nearbyRevealableTiles = tile
+                    .getTilesInDistance(unique.params[2].toInt())
+                    .filter {
+                        !civInfo.exploredTiles.contains(it.position) && it.matchesFilter(
+                            unique.params[1]
+                        )
+                    }
+                    .map { it.position }
+                if (nearbyRevealableTiles.none()) return false
+                civInfo.exploredTiles.addAll(nearbyRevealableTiles
+                    .shuffled(tileBasedRandom)
+                    .apply {
+                        if (unique.params[0] != "All") this.take(unique.params[0].toInt())
+                    }
+                )
+
+                if (notification != null) {
+                    civInfo.addNotification(
+                        notification,
+                        LocationAction(nearbyRevealableTiles.toList())
+                    ) // We really need a barbarian icon
+                }
+
+                return true
+            }
+            "From a randomly chosen tile [] tiles away from the ruins, reveal tiles up to [] tiles away with []% chance" -> {
+                if (tile == null) return false
+                val revealCenter = tile.getTilesAtDistance(unique.params[0].toInt())
+                    .filter { it.position !in civInfo.exploredTiles }
+                    .toList()
+                    .randomOrNull(tileBasedRandom)
+                if (revealCenter == null) return false
+                val tilesToReveal = revealCenter
+                    .getTilesInDistance(unique.params[1].toInt())
+                    .map { it.position }
+                    .filter { tileBasedRandom.nextFloat() < unique.params[2].toFloat() / 100f }
+                civInfo.exploredTiles.addAll(tilesToReveal)
+                civInfo.updateViewableTiles()
+                if (notification != null)
+                    civInfo.addNotification(
+                        notification,
+                        tile.position,
+                        "ImprovementIcons/Ancient ruins"
+                    )
+            }
+            "Triggers voting for the Diplomatic Victory" -> {
                 for (civ in civInfo.gameInfo.civilizations)
                     if (!civ.isBarbarian() && !civ.isSpectator())
-                        civ.addFlag(CivFlags.TurnsTillNextDiplomaticVote.name, civInfo.getTurnsBetweenDiplomaticVotings())
+                        civ.addFlag(
+                            CivFlags.TurnsTillNextDiplomaticVote.name,
+                            civInfo.getTurnsBetweenDiplomaticVotings()
+                        )
+                if (notification != null)
+                    civInfo.addNotification(notification, NotificationIcon.Diplomacy)
+                return true
+            }
+        }
+        return false
+    }
+
+    /** @return boolean whether an action was successfully preformed */
+    fun triggerUnitwideUnique(
```

Comment:
Another totally unimportant naming nitpick: Unitwide'? By belly girth?

Reply:
Was mostly so it sounded similar to `triggerCivwideUnique`, but I can change it of course